src/lib/components/FormChipCell/FormChipCell.js

```diff
@@ -0,0 +1,288 @@
+import React, { useState, useCallback, useRef, useEffect, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import { isEqual } from 'lodash'
+
+import FormChipCellView from './FormChipCellView'
+
+import { isEveryObjectValueEmpty } from '../../utils/common.util'
+import { generateVisibleAndHiddenChipsList } from '../../utils/cutChips'
+import { CHIP_OPTIONS } from '../../types'
+import { CLICK, TAB, TAB_SHIFT } from '../../constants'
+
+const FormChipCell = ({
+  chipOptions,
+  className,
+  delimiter,
+  formState,
+  initialValues,
+  isEditMode,
+  name,
+  onClick,
+  shortChips,
+  visibleChipsMaxLength
+}) => {
+  const [chipsSizes, setChipsSizes] = useState({})
+  const [showHiddenChips, setShowHiddenChips] = useState(false)
+  const [editConfig, setEditConfig] = useState({
+    chipIndex: null,
+    isEdit: false,
+    isKeyFocused: true,
+    isValueFocused: false,
+    isNewChip: false
+  })
+  const [showChips, setShowChips] = useState(false)
+  const [visibleChipsCount, setVisibleChipsCount] = useState(8)
+
+  const chipsCellRef = useRef()
+  const chipsWrapperRef = useRef()
+
+  const handleShowElements = useCallback(() => {
+    if (!isEditMode || (isEditMode && visibleChipsMaxLength)) {
+      setShowHiddenChips(state => !state)
+    }
+  }, [isEditMode, visibleChipsMaxLength])
+
+  let chips = useMemo(() => {
+    return isEditMode || visibleChipsMaxLength === 'all'
+      ? {
+          visibleChips: formState.values[name]
+        }
+      : generateVisibleAndHiddenChipsList(
```

Comment:
can we call it like this?
```suggestion
      : generateChipsList(
```